packages/nextjs/components/scaffold-eth/Drawer.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useState } from "react";
+import { Faucet, NavLinks } from "~~/components/scaffold-eth";
+import RainbowKitCustomConnectButton from "~~/components/scaffold-eth/RainbowKitCustomConnectButton";
+import { Bars3Icon } from "@heroicons/react/24/outline";
+
+/**
+ * Site Drawer
+ */
+export default function Drawer() {
+  const [isOpen, setIsOpen] = useState(Boolean);
+
+  return (
+    <div className="sm:hidden navbar bg-base-100 min-h-0 flex-shrink-0">
+      <div className="navbar-start">
+        <div className="dropdown">
+          <button
+            className="btn btn-ghost ml-1 hover:bg-secondary"
+            // I'm not sure if strictmode is causing this to have inconsistent behavior?
```

Comment:
Are you talking about this ? 

https://user-images.githubusercontent.com/80153681/208154998-83f8364b-92af-4db4-8d9d-724804045c79.mov

As we can see, if we click outside the drawer its closed but the state `IsOpen` is still true and thats why we need to click two times on hamburger icon to open drawer nextime . 

To fix this we you can create a `ref` use it to keep track...checkout out [this](https://www.codingdeft.com/posts/react-on-click-outside/) for reference